WooCommerce/Classes/ViewRelated/Products/ProductSelector/ProductSelectorViewModel.swift

```diff
@@ -178,10 +187,20 @@ final class ProductSelectorViewModel: ObservableObject {
         guard let selectedProduct = products.first(where: { $0.productID == productID }) else {
             return
         }
-        if let onProductSelected = onProductSelected {
-            onProductSelected(selectedProduct)
+
+        if shouldToggleItemOnSelection {
+            // Product multi-selection (new)
+            if let onProductSelected = onProductSelected {
+                toggleSelection(productID: productID)
+                onProductSelected(selectedProduct)
+            }
         } else {
-            toggleSelection(productID: productID)
+            // Product single-selection (legacy)
```

Comment:
Is this legacy being used elsewhere, or can we remove it completely?

Reply:
> Is this legacy being used elsewhere, or can we remove it completely?

This part has been refactored here: https://github.com/woocommerce/woocommerce-ios/pull/8935/commits/976e6e2f4d8e59055c7c34f7ef1dc49ce1d90f50 for easier readability

We currently need it for the case where we use the Product Selector on single item selections, as this will toggle the item without any callback.